nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java

```diff
@@ -161,31 +180,42 @@ protected void init(final ProcessorInitializationContext context) {
     public void setup(final ProcessContext context) {
         String configBody = context.getProperty(MIME_CONFIG_BODY).getValue();
         String configFile = context.getProperty(MIME_CONFIG_FILE).evaluateAttributeExpressions().getValue();
+        String configStrategy = context.getProperty(CONFIG_STRATEGY).getValue();
 
-        if (configBody == null && configFile == null){
-            this.detector = config.getDetector();
-            this.mimeTypes = config.getMimeRepository();
-        } else if (configBody != null) {
-            try {
-                this.detector = MimeTypesFactory.create(new ByteArrayInputStream(configBody.getBytes()));
-                this.mimeTypes = (MimeTypes)this.detector;
-            } catch (Exception e) {
-                context.yield();
-                throw new ProcessException("Failed to load config body", e);
-            }
-
+        if (configBody == null && configFile == null) {
+            setDefaultMimeTypes();
+        } else if (configStrategy.equals(ADD.getValue())) {
+            setDefaultMimeTypes();
+            setCustomMimeTypes(configBody, configFile, context);
         } else {
-            try (final FileInputStream fis = new FileInputStream(configFile);
-                 final InputStream bis = new BufferedInputStream(fis)) {
-                this.detector = MimeTypesFactory.create(bis);
-                this.mimeTypes = (MimeTypes)this.detector;
-            } catch (Exception e) {
-                context.yield();
-                throw new ProcessException("Failed to load config file", e);
-            }
+            setCustomMimeTypes(configBody, configFile, context);
         }
     }
 
+    private void setDefaultMimeTypes() {
+        this.defaultDetector = config.getDetector();
+        this.defaultMimeTypes = config.getMimeRepository();
+    }
+
+    private void setCustomMimeTypes(String configBody, String configFile, ProcessContext context) {
+        try (final InputStream customInputStream = configBody != null ? new ByteArrayInputStream(configBody.getBytes()) : new FileInputStream(configFile)) {
+            this.customDetector = MimeTypesFactory.create(customInputStream);
+            this.customMimeTypes = (MimeTypes) this.customDetector;
+        } catch (Exception e) {
+            context.yield();
+            String configSource = configBody != null ? "body" : "file";
+            throw new ProcessException("Failed to load config " + configSource, e);
+        }
+    }
```

Comment:
Instead of creating multiple MimeTypes, it looks like it should be possible to combine multiple entries use one of several `create()` methods.

Reply:
Done